src/mapreduce.jl

```diff
@@ -87,7 +87,7 @@ function _mapreducedim!(@nospecialize(f), @nospecialize(op), @nospecialize(inito
                         dims::Dims, arrays::Tuple{Vararg{StridedView}})
     if any(isequal(0), dims)
         if !any(isequal(0), size(arrays[1]))
-            map!(initop, arrays[1], arrays[1])
+            isnothing(initop) || map!(initop, arrays[1], arrays[1])
```

Comment:
maybe replace the `if` and the short circuit construction into a single `if`, possbily
```Julia
if length(arrays[1]) != 0 && !isnothing(initop)`
```